### PR TITLE
Allow defaultWorkerPath to be empty

### PR DESCRIPTION
**Pull request by elprans**
`language-executable <file>` should not be only way to launch the
language worker.  For most natively compiled languages the executable by
itself is sufficient. For other languages, like Python, launching a
package as a program is often done using its module path instead of
a file path, e.g. `python -m azure.functions_worker`.  In the latter
case, the launch parameters are passed in "arguments" and
"defaultWorkerPath" is not needed.

### CODE REVIEW

File: src/WebJobs.Script.Grpc/Abstractions/WorkerDescription.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class WorkerDescription
         /// <summary>
         /// Gets or sets the default path to the worker (relative to the bin/workers/{language} directory)
         /// </summary>
-        [JsonProperty(PropertyName = "defaultWorkerPath")]
+        [JsonProperty(PropertyName = "defaultWorkerPath", Required = Required.AllowNull)]
```
Comment (pragnagopa) [inline src/WebJobs.Script.Grpc/Abstractions/WorkerDescription.cs]: Remove required. By default is to allow nulls. Any of these properties are optional as they can be overridden  by a profile. Here is an example: [Java worker.config.json](https://github.com/Azure/azure-functions-java-worker/blob/dev/worker.config.json#L10)

File: src/WebJobs.Script.Grpc/Abstractions/WorkerDescriptionExtensions.cs [hunk 2]
```diff
@@ -21,10 +21,6 @@ public static void Validate(this WorkerDescription workerDescription)
             {
                 throw new ValidationException($"WorkerDescription {nameof(workerDescription.DefaultExecutablePath)} cannot be empty");
             }
```
Comment (pragnagopa) [inline src/WebJobs.Script.Grpc/Abstractions/WorkerDescriptionExtensions.cs]: Add test in [GenericWorkerProviderTests.cs](https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests/Rpc/GenericWorkerProviderTests.cs) that verifies empty/null defaultWorkerPath. Here is the code that builds a test config [GetTestDefaultWorkerDescription](https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests/Rpc/GenericWorkerProviderTests.cs#L374)

**Comment by pragnagopa** [comment]
Can you please share corresponding changes needed in [python worker.config.json](https://github.com/Azure/azure-functions-python-worker/blob/dev/python/worker.config.json)?

**Comment by elprans** [comment]
With these changes python/worker.config.json looks like this:

```json
{
    "description":{
        "language": "python",
        "extensions": [".py"],
        "defaultExecutablePath": "python",
        "arguments": ["-m", "azure.functions_worker"]
    }
}

```

**Comment by elprans** [comment]
Also, `worker.py` is no longer necessary.  If there's a way to omit the `python` directory prefix, then we can ship the `json` file as part of the worker package.

**Comment by pragnagopa** [comment]
> Also, worker.py is no longer necessary. If there's a way to omit the python directory prefix, then we can ship the json file as part of the worker package.
This is a bug. Opened https://github.com/Azure/azure-functions-host/issues/3390 will send a PR soon with the fix.  There is no requirement that folder name should match language

**Comment by pragnagopa** [comment]
@elprans  - Build failed with following
```
Build started, please wait...
Rpc\GenericWorkerProviderTests.cs(126,44): error CS1061: 'IWorkerProvider' does not contain a definition for 'GetWorkerPath' and no accessible extension method 'GetWorkerPath' accepting a first argument of type 'IWorkerProvider' could be found (are you missing a using directive or an assembly reference?) [C:\azure-webjobs-sdk-script\test\WebJobs.Script.Tests\WebJobs.Script.Tests.csproj]
Rpc\GenericWorkerProviderTests.cs(118,117): warning SA1122: Use string.Empty for empty strings [C:\azure-webjobs-sdk-script\test\WebJobs.Script.Tests\WebJobs.Script.Tests.cspro
```
Please update the PR once you address build issue. Thanks.

**Comment by pragnagopa** [comment]
@maiqbal11 @asavaritayal fyi
@elprans - Blocker issue #3390 is now fixed. Rebase and update this PR.

**Comment by elprans** [comment]
@pragnagopa @maiqbal11 Rebased and fixed the tests.  Please review.

**Pull request merged**